### PR TITLE
Add Testing Components

### PR DESCRIPTION
**Pull request by gregdaynes**

### CODE REVIEW

File: hooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env ruby
+
+# require 'rubocop'
+#
```
Comment (brendandeere) [inline hooks/pre-commit]: Not sure if it is useful to commit a precommit hook (particulartly one that is commented out)

File: Gemfile [hunk 2]
```diff
@@ -41,6 +41,8 @@ group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
   gem 'byebug', platform: :mri
   gem 'rspec-rails', '~> 3.5'
+  gem 'pry-rails'
```
Comment (brendandeere) [inline Gemfile]: Probably better to split each gem installation into is own commit. That was each gem installation can be discreet

File: spec/factories.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# spec/factories.rb
+FactoryGirl.define do
+  @password = Faker::Crypto.md5
+
+  factory :person do
```
Comment (brendandeere) [inline spec/factories.rb]: Should be its own commit

File: spec/factories.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    first_name { Faker::Name.first_name }
+    last_name { Faker::Name.last_name }
+    email { Faker::Internet.email }
+    password :password
+    password_confirmation :password
+    date_of_birth { Faker::Date.between(80.year.ago, 18.year.ago) }
+    gender { Faker::Hipster.word }
+  end
+
+  factory :invalid_person, parent: :person do |p|
```
Comment (brendandeere) [inline spec/factories.rb]: Consider using factory girl traits

File: Gemfile [hunk 5]
```diff
@@ -41,6 +41,7 @@ group :development, :test do
...
```
Comment (brendandeere) [inline Gemfile]: Is there a change to the Gemfile lock for thish addition?

File: Gemfile [hunk 6]
```diff
@@ -42,6 +42,7 @@ group :development, :test do
   gem 'byebug', platform: :mri
   gem 'rspec-rails', '~> 3.5'
   gem 'pry-rails'
+  gem 'guard-rspec', require: false
```
Comment (brendandeere) [inline Gemfile]: Is there a change to the Gemfile lock for thish addition?

File: Guardfile [hunk 7]
```diff
@@ -0,0 +1,70 @@
+# A sample Guardfile
+# More info at https://github.com/guard/guard#readme
```
Comment (brendandeere) [inline Guardfile]: A commit should be the smallest atomic, functional piece of code. 

If Guard requires this file than it should be part of the commit where you add guard to the project. Configuration changes can be included later in a separate commit, but only if they are not required for the initial commit to be functional

File: Gemfile [hunk 8]
```diff
@@ -54,5 +54,9 @@ group :development do
   gem 'spring-watcher-listen', '~> 2.0.0'
 end
 
+group :test do
+  gem 'rails-controller-testing'
```
Comment (brendandeere) [inline Gemfile]: Is there a change to the Gemfile lock for thish addition?

File: Gemfile.lock [hunk 9]
```diff
@@ -84,9 +108,19 @@ GEM
...
     minitest (5.10.1)
     multi_json (1.12.1)
+    nenv (0.3.0)
     nio4r (1.2.1)
     nokogiri (1.7.0.1)
       mini_portile2 (~> 2.1.0)
+    notiffany (0.1.1)
+      nenv (~> 0.1)
+      shellany (~> 0.0)
+    pry (0.10.4)
```
Comment (brendandeere) [inline Gemfile.lock]: Many of these changes should be grouped with earlier commits :)

File: spec/factories.rb [hunk 10]
```diff
@@ -0,0 +1,34 @@
+# spec/factories.rb
+FactoryGirl.define do
+  factory :person do
+    first_name 'John'
+    last_name 'Doe'
+    sequence :email do |n|
+      "john.doe.#{}@example.com"
```
Comment (brendandeere) [inline spec/factories.rb]: Im not sure this does what you want

File: Gemfile (hunk 5 above)
Comment (gregdaynes) [inline Gemfile]: It's been squashed to this commit

File: Gemfile (hunk 6 above)
Comment (gregdaynes) [inline Gemfile]: squashed into commit

File: Gemfile (hunk 8 above)
Comment (gregdaynes) [inline Gemfile]: squashed into this commit
